docs/source/our_stack.rst

```diff
@@ -10,31 +10,108 @@ The most up-to-date reference on our codebase is our code itself.
+ends, the callback method is entered:
 
-ROS2
-----
+.. code-block:: python
 
-Vision
-------
+        timer_period = 1 / 60  # seconds
+        self.timer = self.create_timer(timer_period, self.gameplay_tick)
 
-Gameplay
---------
+Motion Control
+--------------
+A lot of the motion control is done on robot by the firmware, but part of the
+motion control is handled by software. See the ``control`` directory,
+``motion_control.cpp`` uses PID to correct robot position. The PID
+controller implementation is not in software, but it is in the
+`fshare repository <https://github.com/RoboJackets/robocup-fshare>`_.
+Correction of robot velocity is done by robot firmware (the code that runs on
+each robot's microcontroller) in the separate firmware repository.
 
 Motion Planning
 ---------------
+Motion planning is about generating a path from a robots current position and
+orientation (pose) to a goal supplied by gameplay. Starting at the lowest
+level, we use RRT for path generation. This occurs after trying to apply a
+straight line path to the goal and that is unsuccessful (if there are no
+obstacles in that straight line why bother with RRT?). We have our own RRT
+implementation in `a separate repository <https://github
+.com/RoboJackets/rrt>`_ which we use as a submodule. This
+way others can easily use our RRT implementation if they want.
+
+But we can't send a path to a robot and expect it to figure it out. We need
+to give our firmware a single velocity command for that instant of planning.
+The file ``soccer/src/soccer/planning/primitives/velocity_profiling.cpp``
+finds that velocity trajectory which we can then return as the current
+trajectory as the robot.
+
+At the highest level of planning, we have several different planners which the
+planning_node selects based on what kind of skill gameplay requested. They
+have different behaviors and based on the their names you can kind of get an
+idea of what each would be used for. They are all located
+in ``soccer/src/soccer/planning/planner``.
+
+This is not an extensive explanation of everything planning does, looking
+through the ``planning`` folder will show you everything else, but it's a
+good start.
+
+Processor
+---------
+It is defined in ``soccer/src/soccer/processor.cpp``, and it basically just
+starts a bunch of temp nodes and then has an infinite loop to update the
+context. Context is used to update the ui. It is good that most other nodes
+are not dependent on this loop.
 
 Radio
 -----
+See the ``/radio`` directory. We have a sim radio for when playing virtual
+matches or testing locally and a network radio for when we run on real
+hardware. It sends commands to our robots using the league designated ports
+for yellow and blue when in sim mode. The simulation packets are standardized
+by the league and in the ``rj_protos`` directory (the ones with the ssl
+prefix).
+
+The network radio expects each robot to know the ip address of where to send
```

Comment:
to send data? or position? A little confusing based on the sentence.

Reply:
The network radio expects the robot to know the ip address of the comp computer. I can clarify.